tests/react/async.test.tsx

```diff
@@ -1143,3 +1144,229 @@ it('multiple derived atoms with dependency chaining and async write (#813)', asy
+    useAtom(derivedSyncAtom)
+    const [count, setCount] = useAtom(countAtom)
+    return { count, setCount, asyncCount }
+  }
+  function TestComponent() {
+    const values = useTest()
+    return <div data-testid="test-comp">{JSON.stringify(values)}</div>
+  }
+  const store = createStore()
+
+  const Wrapper = ({ children }: React.PropsWithChildren) => (
+    <Provider store={store}>{children}</Provider>
+  )
+  render(<TestComponent />, { wrapper: Wrapper })
+
+  const setCount = (arg: SetStateAction<number>) => store.set(countAtom, arg)
+
+  await microtask()
+
+  expect(store.get(countAtom)).toBe(0)
+  expect(store.get(asyncCountAtom)).toBe(0)
+
+  await act(() => setCount(increment))
+  await act(() => setCount(increment))
+
+  resolve[0]!()
+  resolve[1]!()
+  resolve[2]!()
+
+  await macrotask()
+
+  expect(store.get(countAtom)).toBe(2)
+  expect(store.get(asyncCountAtom)).toBe(2)
+
+  await act(() => setCount(increment))
+  await act(() => setCount(increment))
+
+  resolve[3]!()
+  resolve[4]!()
+
+  await macrotask()
+
+  expect(store.get(countAtom)).toBe(4)
+  expect(store.get(asyncCountAtom)).toBe(4) // 3
+})
+
+// TODO: How do we unit test this code with store?
+it('[unit] resolves dependencies reliably after a delay', async () => {
+  const countAtom = atom(0)
+  const asyncCountAtom = atom(0)
+
+  const resolve: (() => void)[] = []
+  const asyncAtom = atom(async (get) => {
+    const count = get(countAtom)
+    await new Promise<void>((r) => resolve.push(r))
+    console.log(`resolved (${count})`)
+    return count
+  })
+
+  const derivedAsyncAtom = atom(
+    async (get, { setSelf }) => {
+      const count = get(countAtom)
+      await microtask()
+      console.log(`derived (${count})`)
+      const asyncCount = await get(asyncAtom)
+      console.log(`derived (${count})`, asyncCount)
+      setSelf(asyncCountAtom, asyncCount)
+    },
+    (_get, set, ...args: Parameters<Setter>) => set(...args)
+  )
+
+  const derivedSyncAtom = atom((get) => {
+    get(derivedAsyncAtom)
+  })
+
+  const store = createStore()
+  store.get(derivedSyncAtom)
+
+  expect(store.get(countAtom)).toBe(0)
+  expect(store.get(asyncCountAtom)).toBe(0)
+
+  store.set(countAtom, increment)
+  store.set(countAtom, increment)
+  await macrotask()
+
+  resolve[0]!()
+  // is it possible to do this test without react hooks?
+  resolve[1]!() // <--- Error: resolve[1] is not defined, it should be.
+  resolve[2]!()
+
+  await microtask()
+  await microtask()
+  await microtask()
+  await macrotask()
+
+  store.get(countAtom)
+  store.get(asyncCountAtom)
+
+  await act(() => store.set(countAtom, increment))
+  await act(() => store.set(countAtom, increment))
+
+  resolve[3]!()
+  resolve[4]!()
+
+  await microtask()
+  await microtask()
+  await microtask()
+  await macrotask()
+
+  expect(store.get(countAtom)).toBe(4)
+  expect(store.get(asyncCountAtom)).toBe(4) // 3
+})
+
+function macrotask() {
+  return new Promise((r) => setTimeout(r, 0))
```

Comment:
```suggestion
  return new Promise((r) => setTimeout(r))
```
Just to be consistent with other tests.